commands.go

```diff
@@ -78,7 +78,7 @@ func processCommand(client Client, payload []byte) error {
 		return client.resetTopics()
 
 	case SetIDKey:
-		if len(blob) != e4crypto.KeyLen {
+		if len(blob) != e4crypto.KeyLen && len(blob) != ed25519.PrivateKeySize {
```

Comment:
I don't have any good ideas how to solve this right now, but I feel like we ought to know which size we require here. It's unlikely we'd end up in an error situation, but still possible.

Reply:
Key size are actually properly checked, it's not possible to set a symkey on a pubkey client or the opposite around. The actual required size checks happen a bit later, when setting the key on the keyMaterial. Here we only ensure  that we got a valid command (so holding either a 32 or 64 bytes key). Then the  key is set on the client's keyMaterial, which is typed as symKey or pubKey, and knows how long its key must be.

Otherwise we can duplicate the size check up there, by exposing the key size or a validation endpoint from the key material on the client, and call it here since we have a client instance available, but I find this less clean involving more code with no extra value.

Or is there any other error case I did miss ?

Reply:
It should be fine, I was just thinking "defence in depth" :)